src/prefect/tasks/secrets/vault_secret.py

```diff
@@ -0,0 +1,161 @@
+        - name (str): The secret name defined by the Vault secret path "<mount-point>/<path>"
+        - vault_credentials_secret (str, optional): name of the PrefectSecret containing Vault
+            credentials.
+                Defaults to a PrefectSecret named `VAULT_CREDENTIALS`.
+                Supported vault client authentication methods:
+                * token { 'VAULT_TOKEN: '<token>' }
+                * appRole: { 'VAULT_ROLE_ID': '<role-id>',
+                             'VAULT_SECRET_ID': '<secret-id>' }
+        - **kwargs (Any, optional): additional keyword args passed to the Task constructor
+
+    Raises:
+        - ValueError: if a `result` keyword is passed
+    """
+
+    def __init__(
+        self, name: str, vault_credentials_secret: str = "VAULT_CREDENTIALS", **kwargs
+    ):
+        self.vault_credentials_secret = vault_credentials_secret
+        kwargs["name"] = name
+        super().__init__(**kwargs)
+
+    def _get_vault_secret(self, name: str):
+        """
+        Connect to vault service and retrieve a secret value
+        Arguments:
+            - name (str): name of the secret to be retrieved using "<mount_point>/<path>"
+
+        Returns:
+            - secret (dict): a dict of the secret key/value items retrieved
+
+        Raises:
+            - ClientError: unable to configure the vault client
+            - PermissionError: unable to authenticate or attempting an unsupported auth method
+            - KeyError: unable to lookup secret in vault using the secret path
+            - PermissionError: provided token/role not authorised to access the secret
+        """
+        self.logger.debug(f"looking up vault path: {name}")
+        client = hvac.Client()
+        # get vault address url
+        vault_url = os.getenv("VAULT_ADDR", default=None)
+        if vault_url is None:
+            vault_url = os.getenv("vault_addr", default=None)
+            if vault_url is None:
+                raise ClientError(
+                    "VAULT_ADDR url var not found. "
+                    'Either "VAULT_ADDR" or "vault_addr" env var required.'
+                )
+        client.url = vault_url
+        self.logger.debug(f"vault addr set to: {client.url}")
+
+        # get vault auth credentials from the PrefectSecret
+        vault_creds = PrefectSecret(self.vault_credentials_secret).run()
+        if "VAULT_TOKEN" in vault_creds.keys():
+            client.token = vault_creds["VAULT_TOKEN"]
+        elif (
+            "VAULT_ROLE_ID" in vault_creds.keys()
+            and "VAULT_SECRET_ID" in vault_creds.keys()
+        ):
+            client.auth_approle(
+                vault_creds["VAULT_ROLE_ID"], vault_creds["VAULT_SECRET_ID"]
+            )
+        else:
+            raise ClientError(
+                "Unable to authenticate with vault service.  "
+                "Supported methods: token, appRole"
+            )
+        if not client.is_authenticated():
+            raise PermissionError(
```

Comment:
I believe `PermissionError` is reserved for file-based permission issues. a `RuntimeError` should suffice here.

Reply:
`PermissionError` exceptions changed to `RuntimeError` exceptions with corresponding docs update.